sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -441,19 +441,16 @@ class ResolveSessionCatalog(
         ShowCreateTableCommand(ident.asTableIdentifier)
       }
 
-    case CacheTableStatement(tbl, plan, isLazy, options) =>
-      val name = if (plan.isDefined) {
-        // CACHE TABLE ... AS SELECT creates a temp view with the input query.
-        // Temp view doesn't belong to any catalog and we shouldn't resolve catalog in the name.
-        tbl
-      } else {
-        parseTempViewOrV1Table(tbl, "CACHE TABLE")
-      }
-      CacheTableCommand(name.asTableIdentifier, plan, isLazy, options)
+    // CACHE TABLE ... AS SELECT creates a temp view with the input query.
+    // Thus, use the identifier in UnresolvedTableOrView directly,
+    case CacheTable(u: UnresolvedTableOrView, plan, isLazy, options) if plan.isDefined =>
+      CacheTableCommand(u.multipartIdentifier.asTableIdentifier, plan, isLazy, options)
```

Comment:
@cloud-fan Please let me know what you think about having `UnresolvedTableOrView` here to eagerly use the identifier if `plan` is defined. Another approach is to have a separate rule to handle `CacheTable(u: UnresolvedTableOrView, ...)`.

Reply:
This is more like CTAS and the `table` should be just `Seq[String]` not `LogicalPlan`.

How about we have both `CacheTable(table: LogicalPlan, ...)` and `CacheTabeAsSelect(tempViewName: String, ...)`?

Reply:
Good idea